### PR TITLE
directory menu: Open in terminal

### PR DESCRIPTION
**Pull request by softwaresale**
In issue lxqt/lxqt#1557, I proposed adding a feature to the directory menu plugin that would allow the user to open a selected terminal. I already have almost all of the code written, so I figured I would submit it. So far, it supports QTerminal and Konsole. I plan on adding support for xterm as well.

### CODE REVIEW

File: plugin-directorymenu/directorymenuconfiguration.ui [hunk 1]
```diff
@@ -40,6 +33,13 @@
         </property>
        </widget>
       </item>
+      <item row="0" column="0">
```
Comment (palinek) [inline plugin-directorymenu/directorymenuconfiguration.ui]: Why did you move this section?

File: plugin-directorymenu/directorymenu.cpp [hunk 2]
```diff
@@ -102,6 +105,13 @@ void DirectoryMenu::openDirectory(const QString& path)
     QDesktopServices::openUrl(QUrl("file://" + QDir::toNativeSeparators(path)));
 }
 
+void DirectoryMenu::openInTerminal(const QString& path)
+{
+    QProcess *process = new QProcess(this);
+    QString action = mDefaultTerminal + " --workdir " + path;
+    process->start(action);
```
Comment (palinek) [inline plugin-directorymenu/directorymenu.cpp]: Here the `process` object leaks...

Comment (softwaresale) [reply plugin-directorymenu/directorymenu.cpp]: Stupid mistake on my part. Fixed

File: plugin-directorymenu/directorymenuconfiguration.ui (hunk 1 above)
Comment (softwaresale) [reply plugin-directorymenu/directorymenuconfiguration.ui]: I believe that this inadvertently happened when I was editing the rest of the file in Qt Creator. I moved that item back to its original spot.

File: plugin-directorymenu/directorymenu.cpp [hunk 3]
```diff
@@ -107,10 +112,10 @@ void DirectoryMenu::openDirectory(const QString& path)
...
 void DirectoryMenu::openInTerminal(const QString& path)
 {
-    QProcess *process = new QProcess(this);
-    QString action = mDefaultTerminal + " --workdir " + path;
-    process->start(action);
-    delete process;
+    QProcess *thisProc = new QProcess(this);
+    QString action = mDefaultTerminal + " -w " + path;
+    mTermRefs.append(thisProc);
+    thisProc->start(action);
```
Comment (palinek) [inline plugin-directorymenu/directorymenu.cpp]: IMO you just need to
```c++
connect(process, &QProcess::finished, process, &QObject::deleteLater);
```
before `start()` and let the pointer "leak", it will tidy up himself.

Or the second posibillity is to use `QProcess::startDetached()` (in this case the child process will not get any SIGHUP signal when the parent (and its terminal) is closed).

**Comment by softwaresale** [comment]
I went ahead and squashed the commits together. I decided to use `QProcess::startDetached`.

**Comment by softwaresale** [comment]
Is there anything that I need to do to finish up this PR?

**Comment by agaida** [comment]
@palinek - can't judge this.

File: plugin-directorymenu/directorymenu.cpp [hunk 4]
```diff
@@ -102,6 +105,12 @@ void DirectoryMenu::openDirectory(const QString& path)
     QDesktopServices::openUrl(QUrl("file://" + QDir::toNativeSeparators(path)));
 }
 
+void DirectoryMenu::openInTerminal(const QString& path)
+{
+    QString action = mDefaultTerminal + " --workdir " + path;
```
Comment (palinek) [inline plugin-directorymenu/directorymenu.cpp]: With this you (or the attacker) can trick the user to execute (malicious) code, right? Please, do not assume that `path` is a nice-looking-no-space-or-other-evil string.

**Comment by softwaresale** [comment]
Do I fix the issue properly? I noticed that `DirectoryMenu::openDirectory` uses `QDir::toNativeSeparators` to check the path, so I did the same.

**Comment by palinek** [comment]
GTM

**Comment by softwaresale** [comment]
Are there any more changes to make? Or is this good to go?

**Comment by palinek** [comment]
For me it's good.
@lxqt/lxqt ?

**Comment by yan12125** [comment]
I like this idea. The implementation might be improved further, and I'm fine to leave it to future pull requests~
